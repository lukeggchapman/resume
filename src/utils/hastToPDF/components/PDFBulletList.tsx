import React from 'react'
import { uid } from 'react-uid'
import { StyleSheet, View, Text } from '@react-pdf/renderer'

const styles = StyleSheet.create({
  column: {
    flexDirection: 'column',
  },
  row: {
    flexDirection: 'row',
    paddingBottom: 2,
  },
  bullet: {
    width: 10,
  },
})

export interface PDFLiProps {
  content: React.ReactNode
}

function PDFLi({ content }: PDFLiProps) {
  return (
    <View style={styles.row}>
      <View style={styles.bullet}>
        <Text>{'\u2022 '}</Text>
      </View>
      <View>{content}</View>
    </View>
  )
}

export interface PDFBulletListProps {
  list: React.ReactNode[]
}

function PDFBulletList({ list }: PDFBulletListProps) {
  return (
    <View style={styles.column}>
      {list.map((listItem) => (
        <PDFLi content={listItem} key={uid(listItem)} />
      ))}
    </View>
  )
}

export { PDFLi }
export default PDFBulletList
