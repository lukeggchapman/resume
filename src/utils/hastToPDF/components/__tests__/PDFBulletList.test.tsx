import React from 'react'
import { shallow } from 'enzyme'

import PDFBulletList, {
  PDFBulletListProps,
  PDFLi,
  PDFLiProps,
} from '../PDFBulletList'

describe('PDFLi', () => {
  const baseProps: PDFLiProps = {
    content: 'ListItem1',
  }
  const render = () => shallow(<PDFLi {...baseProps} />)

  it('renders correctly', () => {
    expect(render()).toMatchSnapshot()
  })
})

describe('PDFBulletList', () => {
  const baseProps: PDFBulletListProps = {
    list: ['ListItem1', 'ListItem2'],
  }
  const render = () => shallow(<PDFBulletList {...baseProps} />)

  it('renders correctly', () => {
    expect(render()).toMatchSnapshot()
  })
})
