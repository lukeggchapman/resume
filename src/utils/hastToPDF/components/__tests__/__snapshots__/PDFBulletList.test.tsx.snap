// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PDFBulletList renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "flexDirection": "column",
      "width": 200,
    }
  }
>
  <PDFLi
    content="ListItem1"
    key="val-ListItem1"
  />
  <PDFLi
    content="ListItem2"
    key="val-ListItem2"
  />
</View>
`;

exports[`PDFLi renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "flex": 1,
      "flexDirection": "row",
      "flexWrap": "wrap",
    }
  }
>
  <View
    style={
      Object {
        "width": 10,
      }
    }
  >
    <Text>
      â€¢ 
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    ListItem1
  </View>
</View>
`;
